src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpSinglelineJavaCheck.java

```diff
@@ -25,23 +25,79 @@
 import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
 
 /**
- * Implementation of a check that looks for a single line in Java files.
- * Supports ignoring comments for matches.
+ * <p>
+ * This class is variation on
+ * <a href="https://checkstyle.org/config_regexp.html#RegexpSingleline">RegexpSingleline</a>
+ * for detecting single lines that match a supplied regular expression in Java files.
+ * It supports suppressing matches in Java comments.
```

Comment:
I don't like that the new first sentence is to say this check is a variation on another check. I think we should explain the check first and then say it is a variation.

Reply:
done